forward/fwd.go

```diff
@@ -177,7 +187,17 @@ func (f *httpForwarder) serveHTTP(w http.ResponseWriter, req *http.Request, ctx
 			stream = contentType == "text/event-stream"
 		}
 	}
-	written, err := io.Copy(newResponseFlusher(w, stream), response.Body)
+	_, err = io.Copy(newResponseFlusher(w, stream), response.Body)
+	if err != nil {
+		ctx.log.Errorf("Error copying upstream response Body: %v", err)
+		ctx.errHandler.ServeHTTP(w, req, err)
+		return
+	}
+
+	response.Body.Close()
```

Comment:
Keeping the defer might be preferable in order to allow future reads of the body inside this method without having to think about adjusting this line. WDYT?